pkg/insights/resyncer.go

```diff
@@ -119,7 +126,68 @@ func meshScoped(t model.ResourceType) bool {
 	return true
 }
 
-func (p *resyncer) resync() error {
+func (p *resyncer) createOrUpdateServiceInsights() error {
+	meshes := &core_mesh.MeshResourceList{}
+	if err := p.rm.List(context.Background(), meshes); err != nil {
+		return err
+	}
+	for _, mesh := range meshes.Items {
+		if need, err := p.needResync(mesh.GetMeta().GetName()); err != nil || !need {
+			continue
+		}
+		err := p.createOrUpdateMeshInsight(mesh.GetMeta().GetName())
+		if err != nil {
+			log.Error(err, "unable to resync resources", "mesh", mesh.GetMeta().GetName())
+			continue
+		}
+	}
+	return nil
+}
+
+func (p *resyncer) createOrUpdateServiceInsight(mesh string) error {
+	insight := &mesh_proto.ServiceInsight{
+		Services: map[string]*mesh_proto.ServiceInsight_DataplaneStat{},
+	}
+	dpList := &core_mesh.DataplaneResourceList{}
+	if err := p.rm.List(context.Background(), dpList, store.ListByMesh(mesh)); err != nil {
+		return err
+	}
+	dpMap := map[model.ResourceKey][]string{}
+	for _, dp := range dpList.Items {
+		dpKey := model.MetaToResourceKey(dp.Meta)
+		for _, inbound := range dp.Spec.Networking.Inbound {
+			svc := inbound.GetService()
+			dpMap[dpKey] = append(dpMap[dpKey], svc)
+			if _, ok := insight.Services[svc]; !ok {
+				insight.Services[svc] = &mesh_proto.ServiceInsight_DataplaneStat{}
+			}
+			insight.Services[svc].Total++
+		}
+	}
+	dpInsights := &core_mesh.DataplaneInsightResourceList{}
+	if err := p.rm.List(context.Background(), dpInsights, store.ListByMesh(mesh)); err != nil {
+		return err
+	}
+	for _, dpInsight := range dpInsights.Items {
+		if dpInsight.Spec.IsOnline() {
+			for _, svc := range dpMap[model.MetaToResourceKey(dpInsight.Meta)] {
+				insight.Services[svc].Online++
+			}
+		}
+	}
+	for _, stat := range insight.Services {
+		stat.Offline = stat.Total - stat.Online
+	}
+	if err := manager.Upsert(p.rm, model.ResourceKey{Mesh: model.NoMesh, Name: mesh}, &core_mesh.ServiceInsightResource{}, func(resource model.Resource) {
+		insight.LastSync = proto.MustTimestampProto(core.Now())
+		_ = resource.SetSpec(insight)
+	}); err != nil {
```

Comment:
I think you need to if when Mesh is deleted like I did with the other one

Reply:
thanks, fixed